src/shortcuts/deleteEmptyThought.js

```diff
@@ -18,13 +19,20 @@ import {
 const canExecute = () => {
   const { cursor, contextViews } = store.getState()
   const offset = window.getSelection().focusOffset
+
   if (cursor) {
     const showContexts = isContextViewActive(contextOf(cursor), { state: store.getState() })
     const contextChain = splitChain(cursor, contextViews)
     const thoughtsRanked = lastThoughtsFromContextChain(contextChain)
-    const children = getThoughtsRanked(thoughtsRanked)
+    const hasChildren = getThoughtsRanked(thoughtsRanked).length > 0
+    const prevThought = getThoughtBefore(cursor)
+    const isAtStart = offset === 0 && !showContexts
+    const hasChildrenAndPrevDivider = prevThought && isDivider(prevThought.value) && hasChildren
 
-    return ((headValue(cursor) === '' && children.length === 0) || isDivider(headValue(cursor))) || (offset === 0 && !showContexts)
+    // delete if the current thought is a divider
+    // delete if the browser selection as at the start of the thought (either deleting or merging if it has children)
+    // do not merge if previous thought is a divider
+    return isDivider(headValue(cursor)) || (isAtStart && !hasChildrenAndPrevDivider)
```

Comment:
Notice how I consolidated the logic into a single statement. Simplification happens through composite boolean variables rather than independent conditions and return statements. Thus the total condition of return can be more easily evaluated.